### DISCUSSION TITLE
Failing to bind devices on my network.

### DISCUSSION THREAD

**Discussion by Erotemic** [Koenkk/zigbee2mqtt#25122] [General]
I'm having an issue with my zigbee network, and I'm hoping to get some help.

All the devices are on the network and connected to my instance of zigbee2mqtt,
but I'm getting a "Failed to bind cluster" error when I try to bind a button
press to toggle a light switch.

My controller is using:

* SONOFF Zigbee 3.0 USB Dongle Plus ZBDongle-E. 

The relevant devices in my zigbee network:

* SONOFF wireless push button (SNZB-01P). Name: "Button 1 - Main Bedroom Light"
* SONOFF smart switch (no neutral) (ZBMINIL2). Name: "Main bedroom wall switch"
* SONOFF occupancy sensor (SNZB-06P). Name: "Occupancy 2"

I have zigbee2mqtt and mqtt running via docker compose. Relevant configuration settings are:

```
serial:
  adapter: ember
advanced:
  homeassistant_legacy_entity_attributes: false
  legacy_api: false
  legacy_availability_payload: false
  channel: 25
  log_level: info
device_options:
  legacy: false
```

Other network details:

```
Zigbee2MQTT version: 1.42.0 commit: 861cba6
Coordinator type: EmberZNet
Coordinator revision: 7.4.4 [GA]
Frontend version: 0.7.6
zigbee-herdsman-converters version: 20.58.0
zigbee-herdsman version: 2.1.9
```

To attempt to bind the button press to toggle the light switch I take the following steps:

* open the zigbee2mqtt frontend web interface
* navigate to devices
* click on "Button 1"
* click on the bind tab
* Configure a new row with:
    * Source endpoint: 1
    * Destination: "Main bedroom wall switch"
    * Destination endpoint: 1
    * Clusters: 
        * OnOff: True
* Click Bind

It waits a second and then the logs throw an error:

```
error 2024-12-07 22:56:30z2m: Failed to bind cluster 'genOnOff' from 'Button 1 - Main Bedroom Light' to 'Main bedroom wall switch' (Error: Bind 0xaddress000000001/1 genOnOff from '0xaddress000000002/1' failed (Delivery failed for '2890'.))
info 2024-12-07 22:56:30z2m:mqtt: MQTT publish: topic 'zigbee2mqtt/bridge/response/device/bind', payload '{"data":{"clusters":[],"failed":["genOnOff"],"from":"Button 1 - Main Bedroom Light/1","to":"Main bedroom wall switch/1"},"error":"Failed to bind","status":"error","transaction":"53tm5-1"}'
error 2024-12-07 22:56:30z2m: Failed to bind
info 2024-12-07 22:56:32zh:ember:ezsp: Received network/route error ROUTE_ERROR_INDIRECT_TRANSACTION_EXPIRY for "2890".
```

When I look at the network map, I have my coordinator connected to "Occupancy 2" with a strength of 29/99, and then "Occupancy 2" is connected to the button with 58 and the switch with 82. 
I have previously tried this when the button was right next to the coordinator and had a very high strength, and I tried to bind it to something else closer by, so I don't think it is a connection issue.

Is there something I can do to diagnose / fix this?

**Comment by Wobak** [comment]
Same issue here. 

Source remote is a E2001/E2002 from IKEA and the cluster is made of TS0505B lightbulbs.

I removed and re-added everything (lightbulbs, group, remote), and I still get that error.

I also changed the joining device, issue remains.

**Comment by Erotemic** [comment]
I've seen some threads that claim some devices don't support binding, so perhaps that is my problem?

https://www.reddit.com/r/homeassistant/comments/rwrsjv/bind_sonoff_button_zigbee2mqtt/

https://github.com/Koenkk/zigbee2mqtt/discussions/15820

I still don't entirely understand the issue.
